stat/roc.go

```diff
@@ -139,17 +139,31 @@ func ROC(cutoffs, y []float64, classes []bool, weights []float64) (tpr, fpr, thr
 //
 // More details about TOC curves are available at
 // https://en.wikipedia.org/wiki/Total_operating_characteristic
-func TOC(classes []bool) (min, ntp, max []float64) {
+func TOC(classes []bool, weights []float64) (min, ntp, max []float64) {
+	if weights != nil && len(classes) != len(weights) {
+		panic("stat: slice length mismatch")
+	}
 	if len(classes) == 0 {
 		return nil, nil, nil
 	}
+
 	ntp = make([]float64, len(classes)+1)
-	for i := range ntp[1:] {
-		ntp[i+1] = ntp[i]
-		if classes[i] {
-			ntp[i+1] += 1
+	if weights == nil {
+		for i := range ntp[1:] {
+			ntp[i+1] = ntp[i]
+			if classes[i] {
+				ntp[i+1] += 1
+			}
+		}
+	} else {
+		for i := range ntp[1:] {
+			ntp[i+1] = ntp[i]
+			if classes[i] {
+				ntp[i+1] += weights[i]
+			}
 		}
```

Comment:
Hah, I just finished doing this, I look back and I see you've done it. Take a look at how I did it and tell me what you think? 
https://github.com/Armadilloa16/gonum/commit/a6176817cab5793242f25706c21bc637951fceac

Reply:
In particular, with weights the calculation of min and max need to be adjusted as well.

Reply:
Yeah, that explains why I didn't understand mine.